commit 807a7f34cd58c537d8b361694d0a4bad23d8aff9
Author: Bill Nottingham <notting@redhat.com>
Date:   Tue Oct 25 15:32:53 2011 -0400

    Don't write to stderr that's not there (or a socket, or whatever, #746263)

diff --git a/service b/service
index 6f17934..2e2d4ea 100755
--- a/service
+++ b/service
@@ -69,6 +69,6 @@ done
 if [ -f "${SERVICEDIR}/${SERVICE}" ]; then
    env -i PATH="$PATH" TERM="$TERM" SYSTEMCTL_IGNORE_DEPENDENCIES=${SYSTEMCTL_IGNORE_DEPENDENCIES} SYSTEMCTL_SKIP_REDIRECT=${SYSTEMCTL_SKIP_REDIRECT} "${SERVICEDIR}/${SERVICE}" ${OPTIONS}
 else
-   echo $"Redirecting to /bin/systemctl ${OPTIONS} ${SERVICE}.service" >/dev/stderr
+   [ -c /dev/stderr ] && echo $"Redirecting to /bin/systemctl ${OPTIONS} ${SERVICE}.service" >/dev/stderr
    exec /bin/systemctl ${OPTIONS} ${SERVICE}.service
 fi
